### PR TITLE
Io2021 c 29/feat/applications management

### PR DESCRIPTION
**Pull request by Petroniuss**
Added endpoints for both managing applications (meaning updating their status) and creating/fetching applications. 
- for employee
image
- for supervisor
image

There's still a place for validating payloads received from the client. 
The preferred way of returning an error is via InvalidRequestException which returns BadRequest with a simple error message - this should be good enough for the front-end.

Also, right now updating the status of an application doesn't add any Delegation/Bonus/Leave - I'd rather discuss how we want to handle that first.

### CODE REVIEW

File: spring-app/src/main/java/com/agh/hr/controllers/employee/ApplicationController.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.agh.hr.controllers.employee;
+
+public class ApplicationController {
+}
```
Comment (Pinioo) [inline spring-app/src/main/java/com/agh/hr/controllers/employee/ApplicationController.java:4]: Not sure if this file has any purpose or is just accidentally added

**Review by Pinioo — CHANGES_REQUESTED** [review]
Everything looks good but I think an endpoint in ApplicationManagementController for fetching single user applications would be useful.

**Comment by ChrisMisan** [comment]
This pull is to be delayed at least to the next meeting as discussed with @Petroniuss

**Comment by Petroniuss** [comment]
I integrated the already discussed code with permissions via this service. I wonder @ChrisFlam3, could we use it in other services? It essentially gives you all users that either are your subordinates or you have permissions to read/modify them (it also takes care of admin case). 

image

I'd like to merge this ASAP so that the front-end could integrate with these endpoints.

**Pull request merged**